blocks/carousel/carousel.js

```diff
@@ -0,0 +1,55 @@
+import {
+  a, div, li, ul, p, img,
+} from '../../scripts/dom-helpers.js';
+
+function apiCardLink(offer) {
+  if ('plucode' in offer) {
+    return `https://www.walgreens.com/store/store/xpo_products.jsp?pluCode=${offer.plucode}`;
+  }
+
+  if ('eventCode' in offer) {
+    return `https://www.walgreens.com/store/BalanceRewardsOffers/balance-rewards-offer.jsp?eventCode=${offer.eventCode}`;
+  }
+
+  // eslint-disable-next-line no-console
+  console.warn('Could not generate link for the following offer', offer);
+  return '';
+}
+
+async function decorateAPICarousel(block) {
+  const apiEndpoint = block.querySelector('a').href;
+  block.innerHTML = '';
+  const apiResponse = await fetch(apiEndpoint);
+
+  if (!apiResponse.ok) {
+    return;
+  }
+
+  const apiInfo = JSON.parse(await apiResponse.text());
+  block.append(
+    ul(
+      ...apiInfo.offers.map((offer) => (
+        li(
+          a({ href: apiCardLink(offer) },
+            div({ class: 'card-image' },
+              img({
+                src: new URL(offer.imageUrl, 'https://www.walgreens.com').toString(),
+                loading: 'lazy',
+                alt: `Offer Image: ${offer.title}`,
+              }),
+            ),
+            div({ class: 'card-body' },
+              p((offer.title)),
+            ),
+          ),
+        )),
+      ),
+    ),
+  );
```

Comment:
same here, maybe instead extracting the card rendering into a function `renderCard`, exporting it in `card.js` and importing it here, or even better `renderCards`, which receives an array of offers and renders the whole list. Sort of decoupling the rendering of a card, from the fact that it is used stand alone vs. in a carousel.